Classes/Plugin/Eid/SearchInDocument.php

```diff
@@ -58,21 +59,37 @@ public function main(ServerRequestInterface $request)
         if ($solr->ready) {
             $query = $solr->service->createSelect();
             $query->setFields([$fields['id'], $fields['uid'], $fields['page']]);
-            $query->setQuery($fields['fulltext'] . ':(' . Solr::escapeQuery((string) $parameters['q']) . ') AND ' . $fields['uid'] . ':' . intval($parameters['uid']));
+            $query->setQuery($this->getQuery($fields, $parameters));
             $query->setStart($count)->setRows(20);
-            $hl = $query->getHighlighting();
-            $hl->setFields([$fields['fulltext']]);
-            $hl->setUseFastVectorHighlighter(true);
-            $results = $solr->service->select($query);
-            $output['numFound'] = $results->getNumFound();
-            $highlighting = $results->getHighlighting();
-            foreach ($results as $result) {
-                $snippet = $highlighting->getResult($result->id)->getField($fields['fulltext']);
+            $query->getHighlighting();
+            $solrRequest = $solr->service->createRequest($query);
+
+            // it is necessary to add the custom parameters to the request
+            // because query object doesn't allow custom parameters
+
+            // field for which highlighting is going to be performed,
+            // is required if you want to have OCR highlighting
+            $solrRequest->addParam('hl.ocr.fl', $fields['fulltext']);
+            // return the coordinates of highlighted search as absolute coordinates
+            $solrRequest->addParam('hl.ocr.absoluteHighlights', 'on');
+            // max amount of snippets for a single page
+            $solrRequest->addParam('hl.snippets', 20);
+
+            $response = $solr->service->executeRequest($solrRequest);
+            $result = $solr->service->createResult($query, $response);
+            $output['numFound'] = $result->getNumFound();
```

Comment:
```suggestion
            /** @scrutinizer ignore-call */
            $output['numFound'] = $result->getNumFound();
```
This is a false-positiv and should be ignored.